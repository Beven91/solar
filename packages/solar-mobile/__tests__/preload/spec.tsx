import Preload from '../../src/preload';

describe('Preload', () => {
  it('show', () => {
    // fade timeout
    jest.useFakeTimers();

    const close = Preload.showLoading('加载中...');

    jest.runOnlyPendingTimers();
    // 断言:渲染内容
    expect(document.body).toMatchSnapshot();

    // 关闭preload
    close();
    // 断言:渲染内容
    expect(document.body).toMatchSnapshot();

    Preload.showLoading('请稍后', false);
    jest.runOnlyPendingTimers();
    // 断言:渲染内容
    expect(document.body).toMatchSnapshot();


    Preload.showLoading('请稍后', false, false);
    // 断言:渲染内容
    expect(document.body).toMatchSnapshot();

    Preload.showLoading('请稍后', true);
    jest.runOnlyPendingTimers();
    // 断言:渲染内容
    expect(document.body).toMatchSnapshot();

    Preload.showLoading(null, true);
    jest.runOnlyPendingTimers();
    // 断言:渲染内容
    expect(document.body).toMatchSnapshot();
  });
});

