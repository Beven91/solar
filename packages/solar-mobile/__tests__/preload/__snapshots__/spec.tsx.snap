// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preload show 1`] = `
<body>
  <div
    id="__preload_container__"
  >
    <div
      class="preload no-mask"
    >
      <span>
        <div
          class="preload-icon default"
        >
          <div
            class="animating"
          />
        </div>
        <div
          class="preload-text"
        >
          加载中...
        </div>
      </span>
    </div>
  </div>
</body>
`;

exports[`Preload show 2`] = `
<body>
  <div
    id="__preload_container__"
  />
</body>
`;

exports[`Preload show 3`] = `
<body>
  <div
    id="__preload_container__"
  >
    <div
      class="preload no-mask"
    >
      <span>
        <div
          class="preload-icon default"
        >
          <div
            class="animating"
          />
        </div>
        <div
          class="preload-text"
        >
          请稍后
        </div>
      </span>
    </div>
  </div>
</body>
`;

exports[`Preload show 4`] = `
<body>
  <div
    id="__preload_container__"
  >
    <div
      class="preload no-mask"
    >
      <span>
        <div
          class="preload-icon default"
        >
          <div
            class="animating"
          />
        </div>
        <div
          class="preload-text"
        >
          请稍后
        </div>
      </span>
    </div>
  </div>
</body>
`;

exports[`Preload show 5`] = `
<body>
  <div
    id="__preload_container__"
  >
    <div
      class="preload am-toast-mask"
    >
      <span>
        <div
          class="preload-icon default"
        >
          <div
            class="animating"
          />
        </div>
        <div
          class="preload-text"
        >
          请稍后
        </div>
      </span>
    </div>
  </div>
</body>
`;

exports[`Preload show 6`] = `
<body>
  <div
    id="__preload_container__"
  >
    <div
      class="preload am-toast-mask"
    >
      <span>
        <div
          class="preload-icon default"
        >
          <div
            class="animating"
          />
        </div>
        
      </span>
    </div>
  </div>
</body>
`;
